### ISSUE TITLE
Make property declarations of CSS rules modifiable

### ISSUE THREAD

**Issue by mganss** [AngleSharp/AngleSharp#297]
Modifying the property declarations of CSS at-rules doesn't seem possible currently. AFAICT this should only apply to `@font-face` and `@viewport`, i.e. those derived from the internal class `CssDeclarationRule`.

For my use case, it would be ideal if the property declarations were exposed as an `ICssStyleDeclaration` property similar to `@page` and `@keyframes` (`ICssPageRule` and `ICssKeyframeRule`).

**Comment by FlorianRappl** [comment]
The `ICssStyleDeclaration` brings A LOT of properties that are not used in this scenario. I think about what would be a good approach.

If you come up with a suggestion like this, could you please elaborate more to allow going into a more detailed discussion? I would love to see code example(s) (what have you tried? how should it look like?) and potential changes (to the API). Also a discussion about alternatives and workarounds would be suitable.

**Comment by mganss** [comment]
My use case is a sanitization library based on AngleSharp: https://github.com/mganss/HtmlSanitizer

We've recently added sanitization of style sheets which necessitates access to at-rules. The code we have for sanitizing property declarations references `ICssStyleDeclaration` because previously we only sanitized inline styles. I was able to reuse this for `ICssStyleRule`, `ICssPageRule`, and `ICssKeyframeRule` but found no way for modifying the property declarations of `ICssFontFaceRule` and `ICssViewportRule`.

I looked at `CssDeclarationRule` and its methods `GetValue` and `SetValue` but it's internal. I also tried modifying the rule's children of type `ICssProperty` but its properties are read-only.

If `ICssStyleDeclaration` is not the way to go, how about letting `CssDeclarationRule` implement an interface `ICssDeclarationRule`:

``` C#
public interface ICssDeclarationRule: IEnumerable<ICssProperty>
{
    void SetProperty(string propertyName, string valueText);
}
```

Ideally, `ICssStyleDeclaration` could inherit that same interface so client code could be kept DRY.

**Comment by FlorianRappl** [comment]
Sounds good. I would propose the following:

``` cs
public interface ICssProperties : IEnumerable<ICssProperty>
{
  String this[String propertyName] { get; }

  [DomName("getPropertyPriority")]
  String GetPropertyPriority(String propertyName);

  [DomName("getPropertyValue")]
  String GetPropertyValue(String propertyName);

  [DomName("setProperty")]
  void SetProperty(String propertyName, String valueText, String priority = null);
}
```

This is a nice basic version of `ICssStyleDeclaration`, such that there is a logical progression.

**Comment by mganss** [comment]
:+1:

**Comment by FlorianRappl** [comment]
Will be released with the next version, see PR #298.
